aws-android-sdk-core/src/main/java/com/amazonaws/logging/LogFactory.java

```diff
@@ -35,29 +35,42 @@
      * @return logger
      */
     public static synchronized Log getLog(Class clazz) {
-        return getLog(clazz.getSimpleName());
+
+        String logTag = clazz.getSimpleName();
+        /** Truncate if length is greater than 23. This is to satisfy the restriction
+        imposed by android on API level 23 through 26 of maximum log tag length of 23. */
+        if (logTag.length() > 23) {
+            android.util.Log.w(TAG, "Truncating log tag length as it exceed 23, the limit imposed by android on certain API Levels");
+            logTag  =logTag.substring(0, 23);
+        }
+        return getLog(logTag);
     }
 
     /**
      * Get the logger for the string tag
-     * @param string the string tag
+     * @param logTag the string tag
      *
      * @return logger
      */
-    public static synchronized Log getLog(final String string) {
-        Log log = logMap.get(string);
+    public static synchronized Log getLog(String logTag) {
+        if (logTag.length() > 23) {
```

Comment:
Consider replacing both of these with a private utility function to `getSafeLogTag` that will truncate (and warn) and could potentially be extended in future to do other sanitizations. That way your code reads more clearly:

```java
String safeTag = getSafeLogTag(logTag); // Or `sanitizeTag` or `normalizeTag`, etc
...
```

Reply:
Done